packages/camelot/src/Camelot.ts

```diff
@@ -2,60 +2,119 @@ import {
   type TransactionFilter,
   type SwapActionParams,
   compressJson,
-  type FilterOperator,
 } from '@rabbitholegg/questdk'
 import { type Address } from 'viem'
 import { CHAIN_ID_ARRAY, ARBITRUM_CHAIN_ID } from './chain-ids'
+import { DEFAULT_TOKEN_LIST } from './contract-addresses'
+import { buildPathQuery, Tokens } from './utils'
+import { CAMELOT_ABI, PARASWAP_ABI } from './abi'
 import {
-  DEFAULT_TOKEN_LIST_URL,
-  WETH_ADDRESS_ARBITRUM,
+  CAMELOT_ROUTER,
+  PARASWAP_ROUTER,
+  ETH_ADDRESS,
 } from './contract-addresses'
-import { CAMELOT_ABI } from './abi'
-import { CAMELOT_ROUTER } from './contract-addresses'
 
-export const buildPathQuery = (tokenIn?: string, tokenOut?: string) => {
-  // v2 paths are formatted as [<token>, <token>]
-  const conditions: FilterOperator[] = []
-
-  if (tokenIn) {
-    conditions.push({ $first: tokenIn })
-  }
-
-  if (tokenOut) {
-    conditions.push({ $last: tokenOut })
-  }
-
-  return {
-    $and: conditions,
-  }
+const isValidContractAddress = (address: Address) => {
+  return (
+    address?.toLowerCase() === CAMELOT_ROUTER.toLowerCase() ||
+    address?.toLowerCase() === PARASWAP_ROUTER.toLowerCase()
+  )
 }
 
 export const swap = async (
   swap: SwapActionParams,
 ): Promise<TransactionFilter> => {
-  const { chainId, contractAddress, tokenIn, tokenOut, amountOut, recipient } =
-    swap
-  const amountIn = tokenIn === WETH_ADDRESS_ARBITRUM ? swap.amountIn : undefined
+  const {
+    chainId,
+    contractAddress,
+    tokenIn,
+    tokenOut,
+    amountIn,
+    amountOut,
+    recipient,
+  } = swap
+
+  const ethUsedIn = tokenIn === Tokens.ETH
+  const ethUsedOut = tokenOut === Tokens.ETH
+
+  if (contractAddress && !isValidContractAddress(contractAddress)) {
+    throw new Error('Invalid Contract Address')
+  }
 
-  // We always want to return a compressed JSON object which we'll transform into a TransactionFilter
   return compressJson({
-    chainId: chainId, // The chainId of the source chain
-    to: contractAddress || CAMELOT_ROUTER, // The contract address of the bridge
+    chainId: chainId,
+    to: { $or: [CAMELOT_ROUTER.toLowerCase(), PARASWAP_ROUTER.toLowerCase()] },
+    value: ethUsedIn ? amountIn : undefined,
```

Comment:
This is a cool pattern - just rely on it getting stripped out if it's not eth based.